core/src/apps/bitcoin/sign_tx/bitcoin.py

```diff
@@ -223,7 +263,7 @@ async def step6_sign_segwit_inputs(self) -> None:
             if i in self.segwit:
                 if i in self.external:
                     txi = await helpers.request_tx_input(self.tx_req, i, self.coin)
-                    self.serialized_tx.extend(txi.witness or b"")
+                    self.serialized_tx.extend(txi.witness or b"\0")
```

Comment:
why?
was this previously a bug?

Reply:
I am not sure whether to call it a bug. If an external segwit input doesn't have a witness, then it's because it had an `ownership_proof`. We can't serialize such a transaction fully, because someone else has to fill in the witnesses of the external inputs. It seems better to put an empty witness (`b"\0"`) here, because then when the coordinator is compiling all the witnesses into the final transaction, it's easier to tell to which input they correspond. It's not entirely necessary though, because the client could tell the coordinator which inputs he signed, but this way the client doesn't have to.